common-client/src/main/scala/org/genivi/sota/client/DeviceRegistryClient.scala

```diff
@@ -106,7 +106,9 @@ class DeviceRegistryClient(baseUri: Uri, devicesUri: Uri, deviceGroupsUri: Uri,
         case StatusCodes.NoContent if ct.runtimeClass == classOf[NoContent] =>
           FastFuture.successful(org.genivi.sota.client.NoContent()).asInstanceOf[Future[T]]
         case other if other.isSuccess() => unmarshaller(response.entity)
-        case err => FastFuture.failed(new Exception(err.toString))
+        case err => log.error(s"Got exception for request to ${httpRequest.uri}\n" +
+          s"Error message: ${response.entity.toString}")
```

Comment:
I made a change so the request body was indeed invalid. The returned 400 error has a json parse message in the body and it displayed fine locally using the code above. It will only display the returned body though, nothing more. I think that is all we need however.